docs/utilities/feature_flags.md

```diff
@@ -3,55 +3,596 @@ title: Feature flags
 
 ## Getting started
+
 ### IAM Permissions
 
-By default, this utility provides AWS AppConfig as a configuration store. As such, you IAM Role needs permission - `appconfig:GetConfiguration` - to fetch feature flags from AppConfig.
+Your Lambda function must have `appconfig:GetConfiguration` IAM permission in order to fetch configuration from AWS AppConfig.
+
+### Required resources
+
+By default, this utility provides [AWS AppConfig](https://docs.aws.amazon.com/appconfig/latest/userguide/what-is-appconfig.html) as a configuration store.
+
+The following sample infrastructure will be used throughout this documentation:
+
+=== "template.yaml"
+
+    ```yaml hl_lines="5 11 18 25 31-50 54"
+    AWSTemplateFormatVersion: "2010-09-09"
+    Description: Lambda Powertools Feature flags sample template
+    Resources:
+      FeatureStoreApp:
+        Type: AWS::AppConfig::Application
+        Properties:
+          Description: "AppConfig Appliction for feature toggles"
+          Name: my-app
+
+      FeatureStoreDevEnv:
+        Type: AWS::AppConfig::Environment
+        Properties:
+          ApplicationId: !Ref FeatureStoreApp
+          Description: "Development Environment for the App Config Store"
+          Name: "development"
+
+      FeatureStoreConfigProfile:
+        Type: AWS::AppConfig::ConfigurationProfile
+        Properties:
+          ApplicationId: !Ref FeatureStoreApp
+          Name: "MyTestProfile"
+          LocationUri: "hosted"
+
+      HostedConfigVersion:
+        Type: AWS::AppConfig::HostedConfigurationVersion
+        Properties:
+          ApplicationId: !Ref FeatureStoreApp
+          ConfigurationProfileId: !Ref FeatureStoreConfigProfile
+          Description: 'A sample hosted configuration version'
+          Content: |
+            {
+                  "premium_features": {
+                    "default": false,
+                    "rules": {
+                      "customer tier equals premium": {
+                        "when_match": true,
+                        "conditions": [
+                          {
+                            "action": "EQUALS",
+                            "key": "tier",
+                            "value": "premium"
+                          }
+                        ]
+                      }
+                    }
+                  },
+                  "feature2": {
+                    "default": true
+                  }
+              }
+          ContentType: 'application/json'
+
+      ConfigDeployment:
+        Type: AWS::AppConfig::Deployment
+        Properties:
+          ApplicationId: !Ref FeatureStoreApp
+          ConfigurationProfileId: !Ref FeatureStoreConfigProfile
+          ConfigurationVersion: !Ref HostedConfigVersion
+          DeploymentStrategyId: "AppConfig.AllAtOnce"
+          EnvironmentId: !Ref FeatureStoreDevEnv
+    ```
+
+=== "CDK"
+
+    ```typescript hl_lines="2-7 13-32 34-35 40 47 54"
+    import * as cdk from '@aws-cdk/core';
```

Comment:
@heitorlessa - i know this might seem be silly, but let's do the CDK example in python too.